classes/class-wc-connect-shipping-method.php

```diff
@@ -262,6 +297,25 @@ private function add_fallback_rate( $service_settings ) {
 		}
 
 		public function calculate_shipping( $package = array() ) {
+			// Skip for carts loaded from session in the dashboard
+			if ( is_admin() && did_action( 'woocommerce_cart_loaded_from_session' ) ) {
+				return;
+			}
+
+			// Skip during Jetpack API requests
+			if ( false !== strpos( $_SERVER['REQUEST_URI'], 'jetpack/v4/' ) ) {
+				return;
+			}
+
+			// Skip during REST API or XMLRPC requests
+			if ( defined( 'REST_REQUEST' ) || defined( 'REST_API_REQUEST' ) || defined( 'XMLRPC_REQUEST' ) ) {
+				return;
+			}
+
+			// Skip during Jetpack REST API proxy requests
+			if ( isset( $_GET['rest_route'] ) && isset( $_GET['_for'] ) && ( 'jetpack' === $_GET['_for'] ) ) {
+				return;
+			}
```

Comment:
Rather than directly copy-paste this code, can we move it somewhere it can be shared, avoiding duplication?